File: dotnet/source/Templates/WebAPI/Devon4Net.WebAPI.Implementation/Business/SearchCriteria/VisitorSearchCriteriaTo.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Devon4Net.WebAPI.Implementation.Business.SearchCriteria
{
    public class VisitorSearchCriteriaTo : AbstractSearchCriteriaTo
    {
        public static long serialVersionUID = 1L;
        public string username { get; set; }
        public string name { get; set; }
        public string phoneNumber { get; set; }
        public string password { get; set; }
        public bool? acceptedCommercial { get; set; }
        public bool? acceptedTerms { get; set; }
        public bool? userType { get; set; }
    }
}

File: java-quarkus2/src/main/java/com/devonfw/application/domain/tos/VisitorSearchCriteriaTo.java
package com.devonfw.application.domain.tos;

import com.devonfw.module.basic.common.api.query.StringSearchConfigTo;

import lombok.Setter;
import lombok.Getter;

/**
 * {@link SearchCriteriaTo} to find instances of
 * {@link com.devonfw.application.visitormanagement.common.api.Visitor}s.
 */
@Getter
@Setter
public class VisitorSearchCriteriaTo extends AbstractSearchCriteriaTo {

  private static final long serialVersionUID = 1L;

  private String username;

  private String name;

  private String phoneNumber;

  private String password;

  private Boolean acceptedCommercial;

  private Boolean acceptedTerms;

  private Boolean userType;

  private StringSearchConfigTo usernameOption;

  private StringSearchConfigTo nameOption;

  private StringSearchConfigTo phoneNumberOption;

  private StringSearchConfigTo passwordOption;
}
